include/minstrobe.hpp

```diff
@@ -0,0 +1,531 @@
+    size_t minstrobe_position_offset{};
+
+    //!\brief Iterator to the first value of minstrobe.
+    urng1_iterator_t urng1_iterator{};
+    //!\brief Iterator to the right most value of the window.
+    urng1_iterator_t second_iterator{};
+    //!\brief Iterator to last element in range.
+    urng1_sentinel_t urng1_sentinel{};
+
+    //!\brief Stored values per window. It is necessary to store them, because a shift can remove the current minstrobe.
+    std::deque<value_t> window_values{};
+    size_t window_size{};
+    bool end{};
+    //!\brief Increments iterator by 1.
+    void next_unique_minstrobe()
+    {
+        while (!next_minstrobe()) {}
+    }
+
+    //!\brief Returns new window value of the first iterator.
+    auto window_value() const
+    {
+        return *urng1_iterator;
+    }
+    
+    //!\brief Returns new window value of the second iterator.
+    auto second_window_value() const
+    {
+        return *second_iterator;
+    }
+
+    //!\brief Advances the window of the first iterator to the next position.
+    void advance_window()
+    {
+        ++urng1_iterator;
+    }
+    
+    //!\brief Advances the window of the second iterator to the next position.
+    void advance_second_window()
+    {
+        ++second_iterator;
+    }
+    
+    //!\brief Calculates minstrobes for the first window.
+    void window_first(size_t const window_min, size_t const window_max)
+    {
+        
+	window_size = ( window_max - window_min + 1 );
+	end = false;
+	
+	if (window_size == 0u)
+            return;
+	second_iterator = urng1_iterator + window_min;
+
+        for (int i = 1u; i < window_size; ++i)
+        {
+            window_values.push_back(second_window_value());
+            advance_second_window();
+        }
+        window_values.push_back(second_window_value());
+
+        auto minstrobe_it = std::ranges::min_element(window_values, std::less_equal<value_t>{});
+        minstrobe_value = std::make_tuple(window_value(), *minstrobe_it);
+        minstrobe_position_offset = std::distance(std::begin(window_values), minstrobe_it);
+	
+    }
+
+    /*!\brief Calculates the next minstrobe value.
+     * \returns True, if new minstrobe is found or end is reached. Otherwise returns false.
+     * \details
+     * For the following windows, we remove the first window value (is now not in window_values) and add the new
+     * value that results from the window shifting.
+     */
+    bool next_minstrobe()
+    {
+    	//for (std::vector<value_type> vec : minstrobes){
+    	//	for (value_type i : vec){ std::cout << i << std::endl;};
+    	//};
+    	advance_second_window();
+    	advance_window();
+        if (second_iterator == urng1_sentinel){
+            end = true;
+            return false;
+        };
+         
+        if (urng1_iterator == urng1_sentinel)
+            return true;
+            
+        if (end) return false;
```

Comment:
I am not sure, what this end does. The only time I see you giving it the value of true is in line 404 and then you return it in line 405. When does it happen that this line 411 is actually used?